File: projects/Grocery-Order-System-Java/Apple.java
/**
 * This class represents apples, which are sold as a countable item.
 * 
 * @author Sanskar Luitel
 * @version Project 10
 * Lab Section Tue/Thur 1:05-2:45PM
 */
public class Apple implements CountableItem {

    private int count;

    public Apple() {
        count = 1;
    }

    public String getName() {
        return "Apple";
    }

    public void setCount(int num) {
        if (num >= 1 && num <= 5) {
            count = num;
        }
    }

    public int getCount() {
        return count;
    }

    public double getUnitPrice() {
        return 0.75;
    }

    public double getFullPrice() {
        return count * getUnitPrice();
    }

    public String toString() {
        return String.format("%s, count: %d, price: $%.2f",
                getName(), getCount(), getFullPrice());
    }
}

File: projects/Grocery-Order-System-CSharp/proj11/Apple.cs
public class Apple : ICountableItem
{
    private int _count = 1;

    public string Name => "Apple";

    public int Count
    {
        get => _count;
        set
        {
            if (value >= 1 && value <= 5)
                _count = value;
        }
    }

    public double UnitPrice => 0.75;

    public double FullPrice => _count * UnitPrice;

    public override string ToString() => $"{Name}, count: {Count}, price: ${FullPrice:F2}";
}
